Non-ascii characters parse incorrectly

When the log file contains non-ascii characters, it seems they are parsed incorrectly.

Say I have a log file with content: 
`[ERROR] [567719dee80742868e17b9414bed686e] 解析脚本异常`

Will be parsed as:

`[ERROR] [567719dee80742868e17b9414bed686e] 解析析析脚脚脚本本本异异异常常常`

--- a/log_viewer/utils.py
+++ b/log_viewer/utils.py
@@ -91,8 +91,12 @@ def readlines_reverse(qfile, exclude=None):
                     line = ""
         else:
             line += next_char
-        char_len = len(next_char.encode())
-        position -= char_len or 1
+
+        char_length = len(next_char.encode())
+        if char_length > 1 and len(line) > 3 and len(set(line[-4:-1])) == 1:
+            line = line[:-3] + next_char
+        position -= char_length or 1
+
     yield line[::-1]
 
 

--- a/log_viewer/utils.py
+++ b/log_viewer/utils.py
@@ -91,7 +91,8 @@ def readlines_reverse(qfile, exclude=None):
                     line = ""
         else:
             line += next_char
-        position -= 1
+        char_len = len(next_char.encode())
+        position -= char_len or 1
     yield line[::-1]
 
 

